commit/tokenprice/types.go

```diff
@@ -0,0 +1,27 @@
+package tokenprice
+
+import (
+	"time"
+
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+)
+
+type Query struct {
+}
+
+type Outcome struct {
+	TokenPrices []cciptypes.TokenPrice `json:"tokenPrices"`
+}
+
+type Observation struct {
+	// FeedTokenPrices for tokens from the feeds on the feed chain
+	FeedTokenPrices []cciptypes.TokenPrice `json:"feedTokenPrices"`
+	// PriceRegistryTokenUpdates for tokens from the PriceRegistry on the dest chain
+	PriceRegistryTokenUpdates []cciptypes.TokenPrice `json:"priceRegistryTokenPrices"`
+	// Observation time
+	Timestamp time.Time
+}
+
+func (o Outcome) Sort() {
+	// TODO: implement
+}
```

Comment:
I don't think this function signature makes sense - is it supposed to mutate the `Outcome` object? If so it needs to be a `o *Outcome` receiver (pointer receiver) rather than value receiver. 

That said I think its less error prone to have a `NewOutcome` method that always guarantees a sorted and deterministic outcome rather than this approach.

Reply:
The `NewOutcome` approach is good as well 👍

Reply:
Anyways let's not change the current approach. Will be better to make in another PR. I'm also going to revert the part where I removed the sort from the encode part.
Created ticket: CCIP-3166

Reply:
I'm not sure I understand, what do you mean by current approach? It seems like this PR changes what was being done in `main`.

Reply:
Not really. `main` already uses sort function in the encode. I just copy pasted it.

Reply:
Right, I see what you mean. I think next step would be to make sure that Outcome objects are deterministic by construction, i.e it is impossible that an Outcome object exists that isn't "sorted" (or similar)